### PR TITLE
Fix compliance refresh_token

### PR DESCRIPTION
**Pull request by jeremymv2**
### Description

This fixes the refresh_token authentication which wasn't working correctly.
It has been tested with ubuntu 14.04 and windows 2012R2

### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

**Comment by andy-dufour** [comment]
Makes sense and LGTM 👍

File: libraries/helper.rb [hunk 1]
```diff
@@ -47,8 +49,8 @@ def with_http_rescue(&block)
   end
 
   # exchanges a refresh token into an access token
-  def retrieve_access_token(server, refresh_token)
-    _success, _msg, access_token = Compliance::API.post_refresh_token(url, refresh_token, options['insecure'])
+  def retrieve_access_token(server_url, refresh_token, insecure = true)
```
Comment (chris-rock) [inline libraries/helper.rb]: Can we set the default for insecure to false here? I think we should assume secure configurations in our code.

File: libraries/profile.rb [hunk 2]
```diff
@@ -61,7 +62,7 @@ class ComplianceProfile < Chef::Resource # rubocop:disable Metrics/ClassLength
 
       # retrieve access token if a refresh token is set
       access_token = token
-      access_token = retrieve_access_token unless refresh_token.nil?
+      access_token = retrieve_access_token(server, refresh_token) unless refresh_token.nil?
```
Comment (chris-rock) [inline libraries/profile.rb]: I would prefer if we would set the `node['audit']['insecure']` attribute as default to `true` and pass in the required attribute.

File: libraries/helper.rb (hunk 1 above)
Comment (chris-rock) [inline libraries/helper.rb]: Lets use false as default and pass in `node['audit']['token']` from the calling function

**Comment by chris-rock** [comment]
@jeremymv2 This is a really good improvement. Just some nitpicking comments :-) LGTM once you addressed the issues

File: libraries/profile.rb (hunk 2 above)
Comment (srenatus) [inline libraries/profile.rb]: @chris-rock: you meant `false`, right? (guessing from the comment above)

Comment (jeremymv2) [inline libraries/profile.rb]: @chris-rock @srenatus I agree completely.  I've added a `node['audit']['insecure']` attribute and set the default to `false` in the library.

**Comment by chris-rock** [comment]
Awesome @jeremymv2 Can you rebase this improvement?

**Comment by chris-rock** [comment]
will be handled with #80

**Pull request closed without merging**